marketplace/marketplace_token_provider.go

```diff
@@ -50,21 +31,27 @@ func (mtp *MarketplaceTokenProvider) RequestToken() (*BearerToken, error) {
 	)
 
 	if err != nil {
-		return nil, fmt.Errorf("could not create the request object: %s", err)
+		logging.Log.Errorf(`error setting up the marketplace token request: %s`, err)
+
+		return nil, errors.New("could not properly prepare to reach the marketplace")
 	}
 
 	// Set the proper headers to accept JSON, and let the server know we're sending urlencoded data.
 	request.Header.Add("Accept", "application/json")
 	request.Header.Add("Content-Type", "application/x-www-form-urlencoded")
 
-	client := GetHttpClient()
+	client := http.Client{Timeout: 3 * time.Second}
```

Comment:
I guess maybe I wasn't clear - rather than having a custom HTTP client I was thinking of just using `http.DefaultClient.Do()`, but then up where `http.NewRequest` is called we can just use `http.NewRequestWithContext`. 

like so: 
```diff
diff --git a/marketplace/marketplace_token_provider.go b/marketplace/marketplace_token_provider.go
index 4bb19c1..d04bd1e 100644
--- a/marketplace/marketplace_token_provider.go
+++ b/marketplace/marketplace_token_provider.go
@@ -1,6 +1,7 @@
 package marketplace
 
 import (
+	"context"
 	"fmt"
 	"net/http"
 	"net/url"
@@ -43,7 +44,12 @@ func (mtp *MarketplaceTokenProvider) RequestToken() (*BearerToken, error) {
 	data.Set("apikey", *mtp.ApiKey)
 	data.Set("grant_type", "urn:ibm:params:oauth:grant-type:apikey")
 
-	request, err := http.NewRequest(
+	// set up a context to control the timeout of the request
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	request, err := http.NewRequestWithContext(
+		ctx,
 		"POST",
 		config.Get().MarketplaceHost+"/api-security/om-auth/cloud/token",
 		strings.NewReader(data.Encode()),
@@ -57,8 +63,7 @@ func (mtp *MarketplaceTokenProvider) RequestToken() (*BearerToken, error) {
 	request.Header.Add("Accept", "application/json")
 	request.Header.Add("Content-Type", "application/x-www-form-urlencoded")
 
-	client := GetHttpClient()
-	response, err := client.Do(request)
+	response, err := http.DefaultClient.Do(request)
 	if err != nil {
 		return nil, fmt.Errorf("could not perform the request to the marketplace: %s", err)
 	}
```